### DISCUSSION TITLE
glTF Working Group Meeting 2022-01-20

### DISCUSSION THREAD

**Discussion by robertlong** [omigroup/gltf-extensions#51] [General]
# OMI glTF Working Group Meeting 1/20/2022

This meeting will be on 1/20/2022 at 10:30 PM (UTC) / 2:30 PM (PST) in the [OMI Discord](https://discord.gg/NJtT9grz5E) within the #omi-weekly-meeting channel. During the meeting, we will be using the #omi-gltf-extensions channel to manage a speaker queue, post links, and for any sidebar discussions.

To be notified of this meeting and others, subscribe to the [OMI Meetings and Events Calendar](https://calendar.google.com/calendar/u/1?cid=Y18wZHB1Z2Y5ZjgzZXE0cWVrbWI2b21xYmptZ0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t) or add yourself to the @omi-gltf-subgroup role in the #roles channel of the OMI Discord.

## Agenda

- OMI_audio_emitter
  - Add Contributors
  - Blender Addon Updates
- Collider extension
- Avatar Animation Retargeting

To propose another item for the agenda, comment below (preferably before the meeting).

**Comment by antpb** [comment]
Notes: mention of avatar humanoid style interop by Discord user tinyCornerstone. Mention of cataloging world by attributes like poly count, texture, filesize, bones, blend shapes, material count, rigging ( rigging is a hard problem )

**Reply by robertlong** [reply]
Spoke extracts some of this info on export for determining scene performance. You can relatively easily extract most of this info from the glTF file. Even in a glb where you can request just the JSON chunk of the file

**Comment by antpb** [comment]
Note on blender addon: Storing binary data from media (like audio/video) is difficult in Blender. May need to start smaller as a first pass. Mentioned in meeting agreement to move forward with glb only as the PR currently sits.

**Comment by antpb** [comment]
Notes Collider:  
Mauve mentioned: define collision boxes and decide physics at a later time.
iFire mentioned: needing rigid, rigid dynamic, custom in the collider spec (may be blocked by deciding approach)
Mauve mentioned: static/dynamic or another that defines collision boxes
Robert mentioned: we should document physics engines and prior work. Document shapes and bodies(rigidbody, softbody, etc)  would be good to document approaches: bullet, NVIDIA PhysX, rapier, godot custom physics


Can we get to stage 4 without object visibility? Robert said yes. 
Lyuma: we need an option to disable shapes or colliders from being actual colliders. Avoid blocking all collisions.


Robert showed two physics engines, Rapier and Unity

Rapier defines a shape of a body using colliders, different shapes available. All gets bundled to collider description which holds the position information. Enables triggers. Finds intersections.

Mauve mentioned: should stay as close to popular engine terminology for searchability relative to different implementations of OMI. fire mentioned we should document those terms from each engine.

Lyuma mentioned: Do bodies require a collision shape to be a child?
